src/registry.rs

```diff
@@ -89,14 +90,14 @@ impl NpmPackageInfo {
 pub struct NpmPackageVersionInfosIterator<'a> {
   iterator: Box<dyn Iterator<Item = &'a NpmPackageVersionInfo> + 'a>,
   info: &'a NpmPackageInfo,
-  newest_dependency_date: Option<chrono::DateTime<chrono::Utc>>,
+  newest_dependency_date: Option<&'a NewestDependencyDateOptions>,
```

Comment:
This is inefficient. We should revert and only store the date so we only need to do the excludes comparison once.